src/block_disagg/block_disagg_write.c

```diff
@@ -111,10 +110,9 @@ __wt_block_disagg_write_internal(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *bloc
 
     /* Get the page ID. */
     page_id = block_meta->page_id;
-
-    /* Get the checkpoint ID. */
     WT_ACQUIRE_READ(checkpoint_id, conn->disaggregated_storage.global_checkpoint_id);
-
+    WT_ASSERT_ALWAYS(session, checkpoint_id == block_meta->checkpoint_id,
```

Comment:
Must these be equal? I don't know this part of the design especially well, but I'd have thought it was possible to increment the global checkpoint ID while writing a page, so the assertion should be that `checkpoint_id >= block_meta->checkpoint_id`.

Reply:
Besides that, I preferred the previous version of this from a readability standpoint - the new version makes it sound like the entire block is about getting the page ID, which isn't true.

Reply:
With the current code, the equality holds. But when we start to increase the checkpoint id, it would be wrong. We need to discuss how to handle that.

Reply:
> Besides that, I preferred the previous version of this from a readability standpoint - the new version makes it sound like the entire block is about getting the page ID, which isn't true.

I don't understand this. Can you be more specific?

Reply:
Discussed this on a call - thanks for explaining the logic and putting the old comments/whitespace back.